### PR TITLE
Improve keyboard navigation in filtering

### LINKED ISSUE #2951: Keyboard navigation issues
### Description  
Keyboard navigation issues

 * igniteui-angular version: 6.2.x
 * browser: all

### Steps to reproduce  

1. Open Grid Moving sample, select the first cell and hold ‘shift + tab’ ; the focus should not propagate to checkbox inside grid header;
2. Open Grid Moving sample, start apply filtering over some column; ‘tab’ to the ‘Cancel’ button, press ‘shift + tab’ - the focus should be moved to the previous button.
3. Open Grid Moving sample, start apply filtering over some column - when reset button is disabled should not be able to focus it with ‘tab’ and ‘shift + tab’
4. Open Grid Moving sample, filter the column 'Phone'; select the first cell in the grid; press ‘shift + tab’; notice that when the focus is on the chip inside "Phone" column and press ‘shift + tab’ again scroll is performed and the focus remains over the chip;

### PR DESCRIPTION
**Pull request by sstoyanovIG**
Closes #2951, #2941

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts [hunk 1]
```diff
@@ -37,14 +37,15 @@ export class IgxFilteringService implements OnDestroy {
...
     private datePipe = new DatePipe(window.navigator.language);
+    private columnStartIndex = -1;
 
     public gridId: string;
     public isFilterRowVisible = false;
     public filteredColumn = null;
     public selectedExpression: IFilteringExpression = null;
-    public columnToChipToFocus = new Map<string, boolean>();
+    public columnToFocus = null;
+    public focusNext = false;
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/filtering/grid-filtering.service.ts]: Perhaps more meaningful name would be something like `shouldFocusNext` or `shouldFocusRight`.

File: projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts [hunk 2]
```diff
@@ -419,8 +420,20 @@ export class IgxGridNavigationService {
                     this.grid.rowEditTabs.last.element.nativeElement.focus();
                     return;
                 }
-                this.navigateUp(currentRowEl, rowIndex,
-                    this.grid.unpinnedColumns[this.grid.unpinnedColumns.length - 1].visibleIndex);
+                if (rowIndex === 0 && this.grid.allowFiltering) {
+                    this.grid.rowList.first.cells.first._clearCellSelection();
+                    const visColLength = this.grid.visibleColumns.length;
+                    if (this.grid.headerContainer.getItemCountInView() === visColLength) {
+                        this.grid.filterCellList.last.focusChip();
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts]: This should be done through the new method in the filtering service.

File: projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-cell.component.ts [hunk 3]
```diff
@@ -309,4 +313,35 @@ export class IgxGridFilteringCellComponent implements AfterViewInit, OnInit {
             this.cdr.detectChanges();
         }
     }
+
+    private ScrollToChip(columnIndex: number, scrollRight: boolean) {
+        this.filteringService.grid.nativeElement.focus({preventScroll: true});
+        this.filteringService.columnToFocus = this.filteringService.grid.visibleColumns[columnIndex];
```
Comment (bkulov) [inline projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-cell.component.ts]: Setting `columnToFocus` and `focusNext` happens only in filtering cell component and in navigation service. And in the both places these two fields are set together and are never read. So in order to make the code a little bit easier to use and error prone we could create a method in the filtering service to set those two fields. They are only used in the service so make them private.

Comment (sstoyanovIG) [reply projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-cell.component.ts]: focusNext is read in the filtering cell when we decide in which direction we should move the focus.

**Comment by npaunov** [comment]
Several scenarios:
1. Pinned Columns, Go to Column Moving Demo. Click Pin ContactName. Click on the first cell in the grid.
  - Click shift and tab. The focus does not move on the last chip.
  - Click on the last chip, hold shift and tab. The focus does not go to the first chip.
2. Grouped Columns. Group a column, click on the first cell of the grid. Click shift + tab. The focus does not move to the last chip.

**Pull request merged**